src/packagedcode/cocoapods.py

```diff
@@ -292,6 +293,96 @@ class PodfileLockHandler(BasePodHandler):
     default_primary_language = 'Objective-C'
     description = 'Cocoapods Podfile.lock'
     documentation_url = 'https://guides.cocoapods.org/using/the-podfile.html'
+    is_lockfile = True
+
+    @classmethod
+    def get_pods_dependency_with_resolved_package(
+        cls,
+        dependency_data,
+        main_pod,
+        dependencies_for_resolved=[],
+    ):
+        """
+        Get a DependentPackage object with its resolved package and
+        dependencies from the `main_pod` string, with additional data
+        populated from the `PodfileLockDataByPurl` mappings.
+        """
+        purl, xreq = parse_dep_requirements(main_pod)
+        base_purl = get_base_purl(purl.to_string())
+
+        resolved_package_mapping = dict(
+            datasource_id=cls.datasource_id,
+            type=cls.default_package_type,
+            primary_language=cls.default_primary_language,
+            namespace=purl.namespace,
+            name=purl.name,
+            version=purl.version,
+            dependencies=dependencies_for_resolved,
+            is_virtual=True,
+        )
+        resolved_package = models.PackageData.from_data(resolved_package_mapping)
+
+        checksum = dependency_data.checksum_by_base_purl.get(base_purl)
+        if checksum:
+            resolved_package.sha1 = checksum
+
+        is_direct = False
+        if base_purl in dependency_data.direct_dependency_purls:
+            is_direct = True
+
+        spec_repo = dependency_data.spec_by_base_purl.get(base_purl)
+        if spec_repo:
+            resolved_package.extra_data["spec_repo"] = spec_repo
+
+        external_source = dependency_data.external_sources_by_base_purl.get(base_purl)
+        if external_source:
+            resolved_package.extra_data["external_source"] = external_source
+
+        return models.DependentPackage(
+            purl=purl.to_string(),
+            # FIXME: why dev?
+            scope='requires',
```

Comment:
where does requires come from?
dependency is what shows in a podspec, like in here https://github.com/nexB/scancode-toolkit/blob/203dec1f6a5d7cac2092ed6904998d24db0f7d74/tests/packagedcode/data/cocoapods/podspec/BadgeHub.podspec#L42 or https://github.com/nexB/scancode-toolkit/blob/203dec1f6a5d7cac2092ed6904998d24db0f7d74/tests/packagedcode/data/cocoapods/podspec/RxDataSources.podspec#L40

See https://guides.cocoapods.org/syntax/podspec.html and this can be qualified with a target as in "ios.dependency"

https://github.com/nexB/scancode-toolkit/blob/203dec1f6a5d7cac2092ed6904998d24db0f7d74/tests/packagedcode/data/cocoapods/podspec.json/FirebaseAnalytics.podspec.json#L5 calls this dependencies

See also https://guides.cocoapods.org/syntax/podfile.html#group_dependencies that talks about dependency scoped by target.

We should reuse this IMHO. This is OK to handle this in another issue to refine Cocoapods support.

Reply:
Note that we could also use this in deplock: https://guides.cocoapods.org/terminal/commands.html#pod_ipc_podfile_json @keshav-space FYI

Reply:
Thanks, this is a seperate issue at https://github.com/nexB/scancode-toolkit/issues/3835

Reply:
> Note that we could also use this in deplock: https://guides.cocoapods.org/terminal/commands.html#pod_ipc_podfile_json

Tracking this here https://github.com/nexB/dependency-inspector/issues/26